### ISSUE TITLE
new models

### ISSUE THREAD

**Issue by daxiongshu** [daxiongshu/Coupon-Purchase-Prediction#5]
Hi, it seems that a file is missing, prefecture_locations.csv, how is it generated?
Hi, I found the file.
Wierd.. It has  'PREF_NAME' in the file, sorry to bother you, let me fix the bug myself.

Fixed, it is a coding issue

**Comment by SudalaiRajkumar** [comment]
Great.. Thanks...

**Comment by SudalaiRajkumar** [comment]
Carl,

I have placed all the codes in github (SRK/Model_V2) with updated readme files. Please let me know if I missed something. 

I will now place the predictions in shared folder and update you. Thank you.

**Comment by SudalaiRajkumar** [comment]
I have created two folders - Test_Preds and Val_Preds in the shared folder. I have updated the test and val predictions files in these folder. Please check the spreadsheet for mapping the files.

To create the submission which scored 0.0107 on LB (sub_newxgb_ens_pp.zip), we need to take Test_Preds/sub_0.0106.7z file from the shared folder and then run 'postProcess_v2.py'. This will create this submission file.  

Please let me know whether this helps. Thank you.

**Comment by daxiongshu** [comment]
Hi, I just generated a validation for sub10. It scored 0.020, which is lower than I expected. Have you tested sub10? Thank you.

**Comment by daxiongshu** [comment]
Hi, I think there is something wrong with my cv for sub10. I appreciate it if you could share me with the correct version of dev_idvs.csv and val_idvs.csv for sub10. thank you very much!

**Comment by SudalaiRajkumar** [comment]
Hi.. I will get the files for sub10 submission. Thank you.

**Comment by SudalaiRajkumar** [comment]
Hi Carl,

I have added the new codes in Model_V3 folder of github. I have created few more variables in this version and this xgb model scored better than the previous xgb model.  

And regarding sub10, it will be really helpful if we don't include it in our bagging. I am not very confident of the dataset used for sub10. These two new datasets (Model_V2 and Model_V3) I am more confident about. So we could bag the models we create using these two datasets.  I will try to add both xgb and rf models in this dataset. It will be really useful if we could run ftrl on these two new datasets and use that as well in our bagging. 

Could you please let me know what are all the constituent models in our best submission? Sorry to bother you much. Thank you.

**Comment by daxiongshu** [comment]
Hi thank you very much!

The best sub on bag16.csv is produced in the following way:
step 1: sub_prob=((sub10_prob+sub_newxgb_ens)/2_0.9+(ftrl1sub.csv)_0.1)_0.8+RF_0.2
step 2: postProcessingv1 sub_prob
step 3: replace 770 users from perauc5.csv

The cv score for this sub is only 0.0245.

My impression is that step 3 adds 0.0003 on public LB. But on the cv, But it decreases cv by 0.0003. And with postProcessing_v2, it doesn't improve public LB.

Another sub bag19.csv has 0.026 cv and 0.01146 lb
step 1: sub_prob=(((sub10_prob+sub_newxgb_ens)/2_0.9+(ftrl1sub.csv)_0.1)_0.8+RF_0.2)_0.5+sub_xgb_dep18_child50_eta05_round450_seed3_trainseed1234_d3_prob_0.5
step 2: postProcessingv1 sub_prob
step 3: replace 770 users from perauc5.csv

So basically neither sub10 nor my per user model improves cv. I guess the reason you don't trust sub10 is that when you change the seed, it becomes a lot worse. So I agree with you that we should trust the ones that do good in both cv and lb.

I am running grid search to blend you new subs now and I believe they can beat our best lb sub and replace them without doubt. 

Let me get back to you when it is done.

Thank you very much.

**Comment by daxiongshu** [comment]
I'm also running a ftrl on modelv3 with a full two-way interaction, however it is very slow. The cv can be done today and if it is good, it takes another day to generate a sub.

**Comment by SudalaiRajkumar** [comment]
Great. Thanks a lot.

By the way, I started exploring NN to add to our mode listl. I hope it will
give a new dimension to our blending. I am able to get around 0.867 val auc
as of now. I will tune them a bit more and add those probability values in
the shared drive.

Thank you.

**Comment by daxiongshu** [comment]
Thank you very much!

Let me run a large bag of xgb and nn on the version3 data.

Best regards,

**Comment by SudalaiRajkumar** [comment]
Carl.. I am back to Chennai.. Thanks a lot for the hard work over the weekend. Let me run few experiments today and let you know if that works. Thank you.. :)
